### PR TITLE
ENG-15048 update DDL compiler

### PR DESCRIPTION
**Pull request by zeemanhe**
fix DDL compiling for TTL export

### CODE REVIEW

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java [hunk 1]
```diff
@@ -1060,26 +1060,35 @@ private Statement readTimeToLive(Table table, boolean alter) {
...
+        if (token.tokenType == Tokens.MAX_FREQUENCY) {
             read();
             if (token.tokenType != Tokens.X_VALUE) {
                 throw unexpectedToken();
             }
-            batchSize = (Integer)(token.tokenValue);
+            maxFrequency = (Integer)(token.tokenValue);
+        }
+
+        if (token.tokenType == Tokens.MIGRATE) {
```
Comment (rdykiel) [inline src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: See above

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java [hunk 2]
```diff
@@ -1060,26 +1060,35 @@ private Statement readTimeToLive(Table table, boolean alter) {
...
+        if (token.tokenType == Tokens.BATCH_SIZE) {
             read();
             if (token.tokenType != Tokens.X_VALUE) {
                 throw unexpectedToken();
             }
-            maxFrequency = (Integer)(token.tokenValue);
+            batchSize = (Integer)(token.tokenValue);
+        }
+
+        if (token.tokenType == Tokens.MIGRATE) {
```
[the remark this answers is not in the document]
Comment (zeemanhe) [reply src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: not a bug. batch size or max_frequency is optional

File: src/frontend/org/voltdb/utils/CatalogSchemaTools.java [hunk 3]
```diff
@@ -111,13 +112,14 @@ public static String toSchema(StringBuilder sb, Table catalog_tbl, String viewQu
...
-            if (isExportOnly) {
+            if (TableType.isStream(catalog_tbl.getTabletype())) {
                 table_sb.append("CREATE STREAM ").append(catalog_tbl.getTypeName());
                 if (streamPartitionColumn != null && viewQuery == null) {
                     table_sb.append(" PARTITION ON COLUMN ").append(streamPartitionColumn);
                 }
                 //Default target means no target.
-                if (streamTarget != null && !streamTarget.equalsIgnoreCase(Constants.DEFAULT_EXPORT_CONNECTOR_NAME)) {
+                if (streamTarget != null && !streamTarget.equalsIgnoreCase(Constants.DEFAULT_EXPORT_CONNECTOR_NAME) &&
+                        TableType.isStream(catalog_tbl.getTabletype())) {
```
[the remark this answers is not in the document]
Comment (zeemanhe) [reply src/frontend/org/voltdb/utils/CatalogSchemaTools.java]: not on road map yet. could deal with it in the future.

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java [hunk 4]
```diff
@@ -1003,7 +1003,7 @@ private Statement compileAlterTableDropTTL(Table t) {
     }
 
     private Statement readTimeToLive(Table table, boolean alter) {
-        //syntax: USING TTL 10 SECONDS ON COLUMN a MAX_FREQUENCY 1 BATCH_SIZE 1000 MIGRATE TO TARGET <TAGRET NAME>
+        //syntax: USING TTL 10 SECONDS ON COLUMN a BATCH_SIZE 1000 MAX_FREQUENCY 1 MIGRATE TO TARGET <TAGRET NAME>
```
[the remark this answers is not in the document]
Comment (zeemanhe) [reply src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: will update it

**Comment by johnTcrawford** [comment]
This is actually for ENG-15408 (not ENG-15048).

**Pull request merged**